### PR TITLE
Fixes dualstack clusterIPs for network policy svc handler

### PR DESCRIPTION
**Pull request by trozet**
Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Review by tssurya — COMMENTED** [review]
/lgtm *if CI passes*

@andreaskaris : you added a bunch of e2e's on the pr where you fixed it in the gateway parts, do you think it makes sense to add policy logic to one of the e2e to make sure we capture this?

**Comment by andreaskaris** [comment]
> /lgtm _if CI passes_
> 
> @andreaskaris : you added a bunch of e2e's on the pr where you fixed it in the gateway parts, do you think it makes sense to add policy logic to one of the e2e to make sure we capture this?

The e2e tests never landed because of the closed window. I'll rebase them and try to get them in this time

File: go-controller/pkg/ovn/gress_policy.go [hunk 1]
```diff
@@ -459,8 +459,9 @@ func getSvcVips(nbClient client.Client, service *v1.Service) []net.IP {
 		}
 	}
 	if util.ServiceTypeHasClusterIP(service) {
-		if util.IsClusterIPSet(service) {
-			ip := net.ParseIP(service.Spec.ClusterIP)
+		ipStrs := util.GetClusterIPs(service)
```
[the remark this answers is not in the document]
Comment (dcbw) [reply go-controller/pkg/ovn/gress_policy.go:462]: I did a quick double-check of other uses of Spec.ClusterIP and I *think* these were the only places that don't handle both ClusterIP + ClusterIPs.

**Comment by dcbw** [comment]
@trozet any idea what the CI failures are?

File: go-controller/pkg/ovn/gress_policy.go [hunk 2]
```diff
@@ -459,8 +459,9 @@ func getSvcVips(nbClient client.Client, service *v1.Service) []net.IP {
...
+		for _, ipStr := range ipStrs {
+			ip := net.ParseIP(ipStr)
 			if ip == nil {
 				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
```
Comment (trozet) [inline go-controller/pkg/ovn/gress_policy.go:466]: oops this is missing a continue, right?

Comment (andreaskaris) [reply go-controller/pkg/ovn/gress_policy.go:466]: Yes, the original code had the issue, too, though:

Original:
~~~
		if util.IsClusterIPSet(service) {
			ip := net.ParseIP(service.Spec.ClusterIP)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			}
			ips = append(ips, ip)
		}
~~~

Should probably have been:
~~~
		if util.IsClusterIPSet(service) {
			ip := net.ParseIP(service.Spec.ClusterIP)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			} else {
			        ips = append(ips, ip)
                        }
		}
~~~

New version:
~~~
		for _, ipStr := range ipStrs {
			ip := net.ParseIP(ipStr)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
			}
			ips = append(ips, ip)
		}
~~~

Should probably be:
~~~
		for _, ipStr := range ipStrs {
			ip := net.ParseIP(ipStr)
			if ip == nil {
				klog.Errorf("Failed to parse cluster IP %q", service.Spec.ClusterIP)
                                continue
			}
			ips = append(ips, ip)
		}
~~~

Comment (dcbw) [reply go-controller/pkg/ovn/gress_policy.go:466]: I just fixed this for Tim and pushed.

**Comment by dcbw** [comment]
Conformance failure for HA/snat/1br/ipv6 is:

```
"[sig-network] NetworkPolicyLegacy [LinuxOnly] NetworkPolicy between server and client should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Networking IPerf2 [Feature:Networking-Performance] should run iperf2",
"[sig-network] Loadbalancing: L7 GCE [Slow] [Feature:NEG] [Flaky] should conform to Ingress spec",
"[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client using UDP should enforce policy based on Ports [Feature:NetworkPolicy]",
"[sig-network] Services should release NodePorts on delete",
"[sig-node] Kubelet when scheduling a read only busybox container should not write to root filesystem [LinuxOnly] [NodeConformance] [Conformance]",
"[sig-node] Kubelet when scheduling a read only busybox container should not write to root filesystem [LinuxOnly] [NodeConformance] [Conformance]",
"[sig-node] Kubelet when scheduling a read only busybox container should not write to root filesystem [LinuxOnly] [NodeConformance] [Conformance]"]}
```

I think that's a much better list than before, so I think this is OK now?

**Pull request merged**